src/instrumentation/initGeneralizationFunctions.ts

```diff
@@ -1,56 +1,78 @@
+  InteracitvityConfiguartion,
+} from 'src/app/regionen/[regionSlug]/_mapData/mapDataSources/sources.const'
 import { prismaClientForRawQueries } from 'src/prisma-client'
-import { Prisma } from '@prisma/client'
 
-// specify license and attribution for data export
-export async function initGeneralizationFunctions(tables) {
+async function createTileSpecification(tableName) {
+  // Get column names and types
+  const columnInformation = await prismaClientForRawQueries.$queryRawUnsafe(`
+  SELECT jsonb_object_agg(column_name, udt_name) - 'geom' - 'minzoom' AS fields
+    FROM information_schema.columns
+    WHERE table_schema = 'public' AND table_name = '${tableName}';`)
+  const { fields } = columnInformation && columnInformation[0] // this object has the form {columnName: columnType}
+  // Get the geometric extent
+  const bbox = await prismaClientForRawQueries.$queryRawUnsafe(
+    `SELECT Array[ST_XMIN(bbox), ST_YMIN(bbox), ST_XMAX(bbox), ST_YMAX(bbox)] as bounds
+    from (
+      SELECT ST_Transform(ST_SetSRID(ST_Extent(geom), 3857), 4326) AS bbox
+        from ${tableName}
+      ) extent;`,
+  )
+  const { bounds } = bbox && bbox[0]
+  // format as vector tile specifaction
+  const tileSpecification = {
+    vector_layers: [{ id: tableName, fields }],
+    bounds,
+  }
+  return tileSpecification
+}
+
+export async function initGeneralizationFunctions(
+  interacitvityConfiguartion: InteracitvityConfiguartion,
+) {
   return Promise.all(
-    tables.map(async (tableName) => {
+    Object.entries(interacitvityConfiguartion).map(async ([tableName, interactivity]) => {
+      // @ts-ignore
       const functionName = generalizationFunctionIdentifier(tableName)
       // Gather meta information for the tile specification
-
-      // Get column names and types
-      const columnInformation = await prismaClientForRawQueries.$queryRawUnsafe(`
-        SELECT jsonb_object_agg(column_name, udt_name) - 'geom' AS fields
-          FROM information_schema.columns
-          WHERE table_schema = 'public' AND table_name = '${tableName}';`)
-      const { fields } = columnInformation && columnInformation[0]
-
-      // Get the geometric extent
-      const bbox = await prismaClientForRawQueries.$queryRawUnsafe(
-        `SELECT ST_XMIN(bbox) AS xmin, ST_YMIN(bbox) AS ymin, ST_XMAX(bbox) AS xmax, ST_YMAX(bbox) AS ymax
-          from (
-            SELECT ST_Transform(ST_SetSRID(ST_Extent(geom), 3857), 4326) AS bbox
-              from ${tableName}
-            ) extent;`,
-      )
-      const { xmin, ymin, xmax, ymax } = bbox && bbox[0]
-      // format as vector tile specifaction
-      const tileSpecification = {
-        vector_layers: [{ id: functionName, fields }],
-        bounds: [xmin, ymin, xmax, ymax],
-      }
+      const tileSpecification = await createTileSpecification(tableName)
+      const { minzoom } = interactivity
+      const stylingKeys = `Array[${interactivity.stylingKeys.map((tag) => `'${tag}'`)}]`
```

Comment:
Finde es immer komisch sich auf dieses default verhalten von array zu verlassen, dass sie kommasepariert in strings überführt werden.
Klarer ist.
```suggestion
      const stylingKeys = `Array[${interactivity.stylingKeys.map((tag) => `'${tag}'`).join(',')}]`
```